### ISSUE TITLE
Rebase the CDK AWS setup on TypeScript?

### ISSUE THREAD

**Issue by roll** [okfn/ckanext-unhcr#251]
# Overview

The CKD's API is so enormously big so it's really hard to write correct code. And they don't have any runtime checks because they rely on static type checking (which is disabled if we use pure JS).

It's a small task but should save a lot of time in the future. 

An example of an error with wich I really stuck with:

```
this.repos.ridl  = codecommit.Repository.fromRepositoryName(this, 'name' ,{
  repositoryName: 'name'
}) // Incorrect

this.repos.ridl  = codecommit.Repository.fromRepositoryName(this, 'name' , 'name') // Correct
```

**Comment by roll** [comment]
I started this one (branch `typescript`) but it seems it still requires some time... day or two or something. 

The worst thing about it that I had to really think hard about stuff like how to pass additional `envName` property and make it work with the existent props type etc. I'm afraid that things like this (absolutely not-related to the problem we solve) can take too much time for relatively simple scripts.

Maybe I just had to start with a fresh head and properly set up VSCode. I don't know. 

Probably still worth to try.
